### DISCUSSION TITLE
Dynamic Textbox

### DISCUSSION THREAD

**Discussion by haithium** [qtile/qtile#4771] [Q&A] [answered]
Hi there, how can I change the value of the `widget.TextBox` in the top bar using cmd? I've tried to create a function, but I don't know how to call it from cmd:

```python
my_message = lambda value: value

message_a = "text 1"
message_b = "text 2"

screens = [
  Screen(
    top=bar.Bar(
      [
        widget.TextBox(text=my_message(message_a)),
        ...
```

**Comment by elParaguayo** [comment]
Create your widget like this:
```python
widget.TextBox(text="text 1", name="textbox1")  # give it a unique name if you have multiple textboxes.
```
To update the text from the command line, do:
```
qtile cmd-obj -o widget textbox1 -f update -a "text 2"
```

**Reply by haithium** [reply]
Can I update the value with a variable name instead of string?

**Reply by elParaguayo** [reply]
What's the trigger for updating the text? Is there a function within your config?

**Reply by haithium** [reply]
I'm trying to customize a bottom bar to simply display a specific keymaps cheatsheet for specific apps when focused, similar to Vim which-key. I have variables with the keys for these apps that contain Pango markup texts in my config. I want to change them when I focus on apps using hooks. So, I'm looking for a way to achieve that. Your response is very helpful, but I want to update the value with a variable that contains the keys cheatsheet.

**Reply by elParaguayo** [reply, marked as answer]
Ok, if you're using a hook, then you can do something like this:
```python
from libqtile import qtile

my_text_1 = "text 1"
my_text_2 = "text 2"

@hook.subscribe....
def my_hooked_func():
    qtile.widgets_map["textbox1"].update(my_text_2)
```

**Reply by haithium** [reply]
Awesome, thanks a lot for your efforts.
